src/main/java/ch/njol/skript/entity/BoatChestData.java

```diff
@@ -34,18 +34,31 @@
 public class BoatChestData extends EntityData<ChestBoat> {
 
 	static {
+		// This ensures all boats are registered
+		// As well as in the correct order via 'ordinal'
+		int boatsSize = Boat.Type.values().length;
+		String[] patterns = new String[boatsSize + 2];
+		patterns[0] = "chest boat";
+		patterns[1] = "any chest boat";
+		for (Boat.Type boat : Boat.Type.values()) {
+			String boatName = boat.toString().replace("_", " ").toLowerCase();
+			if (boatName.equals("bamboo"))
+				boatName = boatName + " chest raft";
+			else
+				boatName = boatName + " chest boat";
+			patterns[boat.ordinal() + 2] = boatName;
+		}
```

Comment:
In the past, if the Locale is not defined, it can return a different value, so Skript standards is to use the ENGLISH or US Locale to match Spigot.
```suggestion
		Boat.Type[] types = Boat.Type.values();
		String[] patterns = new String[types.length + 2];
		patterns[0] = "chest boat";
		patterns[1] = "any chest boat";
		for (Boat.Type boat : types) {
			String boatName = boat.toString().replace("_", " ").toLowerCase(Locale.ENGLISH) + " chest boat";
			if (boatName.startsWith("bamboo"))
				boatName = "bamboo chest raft";
			patterns[boat.ordinal() + 2] = boatName;
		}
```

Same can be done to BoatData